lib/dynamoid/persistence.rb

```diff
@@ -135,7 +135,7 @@ def undump_field(value, options)
                 if value.is_a?(Date) || value.is_a?(DateTime) || value.is_a?(Time)
                   value.to_date
                 else
-                  Time.at(value).to_date
+                  Date.new(1970, 1, 1) + value.to_i
```

Comment:
Let's make `Date.new(1970, 1, 1)` into a constant.  No reason to instantiate it more than once.